### PR TITLE
[PE] fix has_relocations_ field

### PR DESCRIPTION
**Pull request by spobit**

### CODE REVIEW

File: src/PE/Binary.cpp [hunk 1]
```diff
@@ -705,11 +705,13 @@ Binary::it_const_relocations Binary::relocations() const {
 Relocation& Binary::add_relocation(const Relocation& relocation) {
   auto newone = std::make_unique<Relocation>(relocation);
   relocations_.push_back(std::move(newone));
+  this->has_relocations_ = this->relocations_.size();
```
Comment (romainthomas) [inline src/PE/Binary.cpp:708]: I would prefer `.empty()` overs `.size()`

File: src/PE/Binary.cpp [hunk 2]
```diff
@@ -705,11 +705,13 @@ Binary::it_const_relocations Binary::relocations() const {
...
   return *relocations_.back();
 }
 
 void Binary::remove_all_relocations() {
   relocations_.clear();
+  this->has_relocations_ = this->relocations_.size();
```
Comment (romainthomas) [inline src/PE/Binary.cpp:714]: Same

**Review by romainthomas — CHANGES_REQUESTED** [review]
cf. comments

File: src/PE/Binary.cpp (hunk 1 above)
Comment (spobit) [reply src/PE/Binary.cpp]: good idea

**Comment by romainthomas** [comment]
Merged through 4ef1759bd484c37289675cf4d05862d924435f9f (I squashed the commits)

**Pull request closed without merging**